src/main/kotlin/no/nav/syfo/Bootstrap.kt

```diff
@@ -79,198 +104,149 @@ import no.nav.tjeneste.virksomhet.person.v3.informasjon.PersonIdent
-
-fun doReadynessCheck(): Boolean {
-    return true
-}
 
 val objectMapper: ObjectMapper = ObjectMapper()
         .registerModule(JavaTimeModule())
         .registerKotlinModule()
         .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false)
         .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
-val coroutineContext = Executors.newFixedThreadPool(2).asCoroutineDispatcher()
-
-data class ApplicationState(
-    var running: Boolean = true,
-    var ready: Boolean = false
-)
-
-val log = LoggerFactory.getLogger("nav.syfosmmottak-application")!!
+val log: Logger = LoggerFactory.getLogger("no.nav.syfo.syfosmmottak")
 
 const val NAV_OPPFOLGING_UTLAND_KONTOR_NR = "0393"
 
 @KtorExperimentalAPI
-fun main() = runBlocking(coroutineContext) {
+fun main() {
     val env = Environment()
     val credentials = objectMapper.readValue<VaultCredentials>(Paths.get("/var/run/secrets/nais.io/vault/credentials.json").toFile())
     val applicationState = ApplicationState()
+    val applicationEngine = createApplicationEngine(
+            env,
+            applicationState)
 
-    val applicationServer = embeddedServer(Netty, env.applicationPort) {
-        initRouting(applicationState)
-    }.start(wait = false)
+    val applicationServer = ApplicationServer(applicationEngine)
+    applicationServer.start()
 
     DefaultExports.initialize()
 
-    connectionFactory(env).createConnection(credentials.mqUsername, credentials.mqPassword).use { connection ->
-        connection.start()
-
-        Jedis(env.redishost, 6379).use { jedis ->
-            val session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE)
+    val kafkaBaseConfig = loadBaseConfig(env, credentials)
 
-            val inputconsumer = session.consumerForQueue(env.inputQueueName)
-            val syfoserviceProducer = session.producerForQueue(env.syfoserviceQueueName)
-            val backoutProducer = session.producerForQueue(env.inputBackoutQueueName)
+    val producerProperties = kafkaBaseConfig.toProducerConfig(env.applicationName, valueSerializer = JacksonKafkaSerializer::class)
 
-            val kafkaBaseConfig = loadBaseConfig(env, credentials)
+    val kafkaproducerreceivedSykmelding = KafkaProducer<String, ReceivedSykmelding>(producerProperties)
 
-            val producerProperties = kafkaBaseConfig.toProducerConfig(env.applicationName, valueSerializer = JacksonKafkaSerializer::class)
+    val kafkaproducervalidationResult = KafkaProducer<String, ValidationResult>(producerProperties)
 
-            val kafkaproducerreceivedSykmelding = KafkaProducer<String, ReceivedSykmelding>(producerProperties)
+    val kafkaproducerApprec = KafkaProducer<String, Apprec>(producerProperties)
 
-            val kafkaproducervalidationResult = KafkaProducer<String, ValidationResult>(producerProperties)
+    val manualTaskproducerProperties = kafkaBaseConfig.toProducerConfig(env.applicationName, valueSerializer = KafkaAvroSerializer::class)
```

Comment:
Kan vi bruke oppgave i stedet for task, hvis det er det vi mener?

Reply:
jepp